pkg/operator/controller/externaldns/deployment.go

```diff
@@ -72,15 +75,26 @@ var sourceStringTable = map[operatorv1alpha1.ExternalDNSSourceType]string{
 // ensureExternalDNSDeployment ensures that the externalDNS deployment exists.
 // Returns a Boolean value indicating whether the deployment exists, a pointer to the deployment, and an error when relevant.
 func (r *reconciler) ensureExternalDNSDeployment(ctx context.Context, namespace, image string, serviceAccount *corev1.ServiceAccount, externalDNS *operatorv1alpha1.ExternalDNS) (bool, *appsv1.Deployment, error) {
-
 	nsName := types.NamespacedName{Namespace: namespace, Name: controller.ExternalDNSResourceName(externalDNS)}
-
-	secretName := controller.ExternalDNSDestCredentialsSecretName(r.config.Namespace, externalDNS.Name).Name
 	configMapName := ""
 	if r.config.InjectTrustedCA {
 		configMapName = controller.ExternalDNSDestTrustedCAConfigMapName("").Name
 	}
-	desired, err := desiredExternalDNSDeployment(namespace, image, secretName, serviceAccount, externalDNS, r.config.IsOpenShift, r.config.PlatformStatus, configMapName)
+
+	secretExists, secret, err := r.currentExternalDNSSecret(ctx, nsName)
```

Comment:
This will not work, deployment name != secret name.
```suggestion
	secretExists, secret, err := r.currentExternalDNSSecret(ctx, controller.ExternalDNSDestCredentialsSecretName(namespace, externalDNS.Name))
```
I think that e2e tests failed because of this: no secret was found - no externaldns operand was created.